scripts/install-appstudio-kcp.sh

```diff
@@ -0,0 +1,220 @@
+    echo -e "   # Deploy Red Hat App Studio in kcp unstable version"
+    echo -e "   /bin/bash <E2E_DIR>/scripts/install-appstudio-kcp.sh -kc kcp-unstable-root -kk <path-to-kcp-kubeconfig> -ck <path-to-openshift-cluster-kubeconfig>\n"
+    echo -e "To authenticate against a KCP instance using Red Hat SSO before starting you need to get an offline token from https://console.redhat.com/openshift/token\n"
+    echo -e "   export OFFLINE_TOKEN=<token_goes_here>"
+}
+
+while [[ $# -gt 0 ]]
+do
+    case "$1" in
+        -h|--help)
+            helpUsage
+            exit 0
+            ;;
+        -kc|--kcp-context)
+            export KCP_CONTEXT=$2
+            ;;
+        -kk|--kcp-kubeconfig)
+            export KCP_KUBECONFIG=$2
+            echo $2
+            
+            ;;
+        -ck|--cluster-kubeconfig)
+            export CLUSTER_KUBECONFIG=$2
+            ;;
+        -s|--stable)
+            export IS_STABLE="true"
+            ;;
+        *)
+            ;;
+    esac
+    shift
+done
+
+if [[ -z "$KCP_CONTEXT" ]]; then
+    echo "[ERROR] Not KCP context defined in the script. Please use flag '-kc' or '--kcp-context' to define the kcp context." 
+    helpUsage & exit 1
+fi
+
+if [[ -z "$KCP_KUBECONFIG" ]]; then
+    echo "[ERROR] KCP kubeconfig not defined. Please use flag '-kk' or '--kcp-kubeconfig' to define the kcp kubeconfig." 
+    helpUsage & exit 1
+fi
+
+if [[ -z "$CLUSTER_KUBECONFIG" ]]; then
+    echo "[ERROR] Not cluster kubeconfig defined. Please use flag '-ck' or '--cluster-kubeconfig' to define the kcp physical cluster target kubeconfig." 
+    helpUsage & exit 1
+fi
+
+if [[ -z "$OFFLINE_TOKEN" ]]; then
+    echo "[ERROR] OFFLINE_TOKEN environment is not set. You can obtain one from cloud.redhat.com." 
+    helpUsage & exit 1
+fi
+
+# Export kcp kubeconfig to start working against KCP
+export KUBECONFIG="$KCP_KUBECONFIG"
+
+# Installing oidc-login plugin for kubectl. More information about oidc-login plugin can be found here: https://github.com/int128/kubelogin.
+# oidc-login will be used to authenticate using SSO against KCP
+function installKubectlOIDCLoginPlugin() {
+    echo -e "[INFO] Installing krew for oidc-login plugin installation."
+    (
+        set -x; cd "$(mktemp -d)" &&
+        OS="$(uname | tr '[:upper:]' '[:lower:]')" &&
+        ARCH="$(uname -m | sed -e 's/x86_64/amd64/' -e 's/\(arm\)\(64\)\?.*/\1\2/' -e 's/aarch64$/arm64/')" &&
+        KREW="krew-${OS}_${ARCH}" &&
+        curl -fsSLO "https://github.com/kubernetes-sigs/krew/releases/latest/download/${KREW}.tar.gz" &&
+        tar zxvf "${KREW}.tar.gz" &&
+        ./"${KREW}" install krew
+    )
+    export PATH="${KREW_ROOT:-$HOME/.krew}/bin:$PATH"
+    kubectl krew install oidc-login
+}
+
+# Will install all kcp plugins used by Red Hat App Studio installation script
+function installKubectlKcpPlugins() {
+    local kcp_clone_branch="main"
```

Comment:
should there be a code for checking whether plugins are not already installed?

Reply:
Not necessary. Better to update the plugins..

Reply:
not sure why would plugins need to be updated every time, even in case when the required version of plugins is already installed. for example when I'm running this script locally (not for the first time)

Reply:
done :)